numpy/core/tests/test_deprecations.py

```diff
@@ -533,3 +533,9 @@ def test_get_numeric_ops(self):
         # other tests.
         self.assert_deprecated(np.set_numeric_ops, kwargs={})
         assert_raises(ValueError, np.set_numeric_ops, add='abc')
+
+
+class TestShape1Fields(_DeprecationTestCase):
+    # 2019-05-20, 1.17.0
+    def test_shape_1_fields(self):
+        assert_warns(FutureWarning, np.dtype, '1u1')
```

Comment:
Should use self.assert_deprecated which checks twice, once for logging and once for raising (since those are different c code paths).

Reply:
done